src/common/transformations/src/transformations/common_optimizations/sdpa_fusion.cpp

```diff
@@ -73,37 +157,76 @@ SDPAFusion::SDPAFusion() {
             pattern_map.at(softmax).get_target_inputs().size() > 1) {
             return false;
         }
-        if (pattern_map.count(optional_add_mask) && (pattern_map.at(optional_add_mask).get_target_inputs().size() > 1 ||
+         if (pattern_map.count(qk_opt_scaled_mask_added) && (pattern_map.at(qk_opt_scaled_mask_added).get_target_inputs().size() > 1 ||
                                                      pattern_map.at(mask).get_partial_shape().size() > 4)) {
             return false;
         }
 
-        Output<ov::Node> mask_value;
-        Output<ov::Node> mask_input;
-        if (pattern_map.find(optional_add_mask) != pattern_map.end()) {
-            mask_value = pattern_map.at(mask);
+        ov::Output<ov::Node> scale_node;
+        if (pattern_map.count(attn_scale) > 0) {
+            scale_node = pattern_map.at(attn_scale);
+            auto attn_scale_out_ps = scale_node.get_partial_shape();
+            if (attn_scale_out_ps.is_dynamic())
+                return false;
+            // attn_scale layer should have only single output scalar value
+            if (ov::shape_size(attn_scale_out_ps.get_shape()) != 1)
+                return false;
+            // we need to be able to cast attn_scale layer to Constant layer
+            // in order to read actual scale value
+            auto attn_scale_const_m = ov::as_type_ptr<ov::op::v0::Constant>(scale_node.get_node_shared_ptr());
+            if (!attn_scale_const_m)
+                return false;
+            auto attn_scale_val_ptr = static_cast<const void*>(attn_scale_const_m->get_data_ptr());
+            scale_node = ov::op::v0::Constant::create(T, ov::Shape{}, attn_scale_val_ptr);
         } else {
-            mask_value = ov::op::v0::Constant::create(q_node.get_element_type(), ov::Shape{}, std::vector<float>{0});
+            scale_node = ov::op::v0::Constant::create(T, ov::Shape{}, {1.0});
         }
 
-        if (mask_value.get_partial_shape().size() > 4) {
-            return false;
-        }
-
-        if (mask_value.get_partial_shape().rank() == 0 || mask_value.get_partial_shape().rank() == 4) {
-            mask_input = mask_value;
+        Output<ov::Node> mask_input;
+        if (pattern_map.count(mask) > 0) {
+            // for some reason line below doesn't work for all cases,
+            // so need to explicitly point to correct qk layer
+            // auto qk_out = pattern_map.at(qk_opt_scaled_pre_mask_opt_reshaped);
+            ov::Output<ov::Node> qk_out;
+            if (pattern_map.count(qk_opt_scaled_pre_mask_reshaped) > 0)
+                qk_out = pattern_map.at(qk_opt_scaled_pre_mask_reshaped);
+            else if (pattern_map.count(qk_unsqueeze) > 0)
+                qk_out = pattern_map.at(qk_unsqueeze);
+            else if (pattern_map.count(qk) > 0)
+                qk_out = pattern_map.at(qk);
+            else
+                return false;
```

Comment:
Probably can be simplified by taking first input of `qk_opt_scaled_mask_added` node.